cli/containerdutils/containerd.go

```diff
@@ -0,0 +1,83 @@
+package containerdutils
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/docker/api/types"
+	registrytypes "github.com/docker/docker/api/types/registry"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/docker/distribution/reference"
+)
+
+// IsContainerdPresent very simple routine to check to see if the containerd socket is present
+func IsContainerdPresent() bool {
+	// Check for presence first
+	_, err := os.Stat(ContainerdSockPath)
+	return err == nil
+}
+
+// GetContainerdClient verifies the containerd socket is present and writable by the current user
+func GetContainerdClient() (*containerd.Client, error) {
+	// Check for presence first
+	_, err := os.Stat(ContainerdSockPath)
```

Comment:
tisk tisk ;), connect and handle errors on connect os.IsNotExist, not check first.  These types of checks are racy